rust/connlib/tunnel/src/peer.rs

```diff
@@ -4,14 +4,113 @@ use std::time::Instant;
 
 use bimap::BiMap;
 use chrono::{DateTime, Utc};
-use connlib_shared::messages::{ClientId, DnsServer, GatewayId, ResourceId};
+use connlib_shared::messages::{
+    gateway::Filter, gateway::Filters, ClientId, DnsServer, GatewayId, ResourceId,
+};
 use ip_network::IpNetwork;
 use ip_network_table::IpNetworkTable;
-use ip_packet::{MutableIpPacket, Packet};
+use ip_packet::ip::IpNextHeaderProtocols;
+use ip_packet::{IpPacket, MutableIpPacket, Packet};
+use rangemap::RangeInclusiveSet;
 
 use crate::client::IpProvider;
+use crate::utils::contains;
 
-type ExpiryingResource = (ResourceId, Option<DateTime<Utc>>);
+#[derive(Debug)]
+enum FilterEngine {
+    PermitAll,
+    PermitSome(FilterEngineInner),
+}
+
+#[derive(Debug)]
+struct FilterEngineInner {
+    udp: RangeInclusiveSet<u16>,
+    tcp: RangeInclusiveSet<u16>,
+    icmp: bool,
+}
+
+impl From<&Filters> for FilterEngine {
+    fn from(value: &Filters) -> Self {
+        FilterEngine::PermitSome(value.into())
+    }
+}
+
+impl FilterEngine {
+    fn new() -> FilterEngine {
```

Comment:
Is `new` a good name here? Why is `PermitSome` even the default? Shouldn't `PermitAll` be the default?